### PR TITLE
BXMSPROD-1741 Openshift Serverless Logic nightly

### PR DESCRIPTION
**Pull request by Ginxo**
https://issues.redhat.com/browse/BXMSPROD-1741

* https://gitlab.cee.redhat.com/middleware/build-configurations/-/merge_requests/847
* https://gitlab.cee.redhat.com/business-automation/eng-jenkins/-/merge_requests/168
* https://github.com/kiegroup/kie-jenkins-scripts/pull/1265
* https://github.com/kiegroup/kogito-pipelines/pull/506

To be merged after https://github.com/kiegroup/kie-jenkins-scripts/pull/1264

<pre>
How to retest a PR or trigger a specific build:

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
</pre>

### CODE REVIEW

**Comment by mareknovotny** [comment]
jenkins retest this please

File: job-dsls/jobs/prod/prod_cron_meta_nightly_pipeline.groovy [hunk 1]
```diff
@@ -173,6 +179,22 @@ String kogitoWithSpecDroolsNightlyStage(String kogitoVersion, String kogitoBranc
     """
 }
 
+String serverlessLogicNightlyStage(String productVersion, String branch, String configBranch, String nexusSuffix = getNexusFromVersion(productVersion)) {
+    return """
+        stage('trigger KOGITO nightly job ${productVersion}') {
```
Comment (radtriste) [inline job-dsls/jobs/prod/prod_cron_meta_nightly_pipeline.groovy:185]:
```suggestion
        stage('trigger Serverless Logic nightly job ${productVersion}') {
```

File: job-dsls/jobs/prod/prod_cron_meta_nightly_pipeline.groovy [hunk 2]
```diff
@@ -173,6 +179,22 @@ String kogitoWithSpecDroolsNightlyStage(String kogitoVersion, String kogitoBranc
...
+            steps {
+                build job: 'kogito.nightly/${branch}', propagate: false, wait: true, parameters: [
```
Comment (radtriste) [inline job-dsls/jobs/prod/prod_cron_meta_nightly_pipeline.groovy:187]: should we rename the nightly job to `serverless-logic.nightly` ?

Comment (mareknovotny) [reply job-dsls/jobs/prod/prod_cron_meta_nightly_pipeline.groovy:187]: i would still keep there kogito, so kogito-swf.nightly?

Comment (Ginxo) [reply job-dsls/jobs/prod/prod_cron_meta_nightly_pipeline.groovy:187]: I would too, do you agree @radtriste

Comment (radtriste) [reply job-dsls/jobs/prod/prod_cron_meta_nightly_pipeline.groovy:187]: ok

File: job-dsls/jobs/prod/prod_cron_meta_nightly_pipeline.groovy [hunk 3]
```diff
@@ -173,6 +180,23 @@ String kogitoWithSpecDroolsNightlyStage(String kogitoVersion, String kogitoBranc
     """
 }
 
+String serverlessLogicNightlyStage(String productVersion, String droolsVersion, String branch, String configBranch, String nexusSuffix = getNexusFromVersion(productVersion)) {
+    return """
+        stage('trigger Serverless Logic nightly job ${productVersion}') {
+            steps {
+                build job: 'kogito.nightly/${branch}', propagate: false, wait: true, parameters: [
+                        [\$class: 'StringParameterValue', name: 'DEPLOYMENT_REPO_URL', value: 'https://bxms-qe.rhev-ci-vms.eng.rdu2.redhat.com:8443/nexus/service/local/repositories/scratch-release-kogito-${nexusSuffix}/content-compressed'],
```
Comment (winklerm) [inline job-dsls/jobs/prod/prod_cron_meta_nightly_pipeline.groovy:188]: I think we need to use a different Nexus repository.

File: job-dsls/jobs/prod/prod_cron_meta_nightly_pipeline.groovy [hunk 4]
```diff
@@ -173,6 +180,23 @@ String kogitoWithSpecDroolsNightlyStage(String kogitoVersion, String kogitoBranc
...
+                        [\$class: 'StringParameterValue', name: 'UMB_VERSION', value: '${getUMBFromVersion(productVersion)}'],
```
Comment (winklerm) [inline job-dsls/jobs/prod/prod_cron_meta_nightly_pipeline.groovy:189]: The UMB message needs to be changed so we do not send the same message by two jobs - or is it expected that kogito 1.24 would be only for serverless logic?

File: job-dsls/jobs/prod/prod_cron_meta_nightly_pipeline.groovy (hunk 3 above)
Comment (winklerm) [reply job-dsls/jobs/prod/prod_cron_meta_nightly_pipeline.groovy:188]: Or just `scratch-release-kogito-swf-1.24` would do?

File: job-dsls/jobs/prod/prod_cron_meta_nightly_pipeline.groovy (hunk 4 above)
Comment (mareknovotny) [reply job-dsls/jobs/prod/prod_cron_meta_nightly_pipeline.groovy:189]: yes, and some future followed up releases too

File: job-dsls/jobs/prod/prod_cron_meta_nightly_pipeline.groovy (hunk 3 above)
Comment (winklerm) [reply job-dsls/jobs/prod/prod_cron_meta_nightly_pipeline.groovy:188]: I see - if the Kogito 1.24 is actually not specific to SWF (I mean if there will be just "one" Kogito use by SWF and whatever else), we could just leave scratch-release-kogito-1.24.

My original concern was about a potential conflict between a SWF Kogito and Kogito for other use-cases.

Comment (Ginxo) [reply job-dsls/jobs/prod/prod_cron_meta_nightly_pipeline.groovy:188]: can we resolve this one @winklerm ?

File: job-dsls/jobs/prod/prod_cron_meta_nightly_pipeline.groovy (hunk 4 above)
Comment (Ginxo) [reply job-dsls/jobs/prod/prod_cron_meta_nightly_pipeline.groovy]: `-blue` suffix added to `UMB_VERSION`

File: job-dsls/jobs/prod/prod_cron_meta_nightly_pipeline.groovy (hunk 3 above)
Comment (winklerm) [reply job-dsls/jobs/prod/prod_cron_meta_nightly_pipeline.groovy:188]: Agreed, let's resolve it.

**Pull request merged**